vortexor/src/load_balancer.rs

```diff
@@ -0,0 +1,193 @@
+//! Module responsible for interacting with RPC services.
+//! Adapted from jito-replayer code.
+
+use {
+    crossbeam_channel::{Receiver, RecvTimeoutError, Sender},
+    dashmap::DashMap,
+    log::{error, info},
+    solana_client::{pubsub_client::PubsubClient, rpc_client::RpcClient},
+    solana_metrics::{datapoint_error, datapoint_info},
+    solana_sdk::{
+        clock::Slot,
+        commitment_config::{CommitmentConfig, CommitmentLevel},
+    },
+    std::{
+        sync::{
+            atomic::{AtomicBool, AtomicU64, Ordering},
+            Arc,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
+        time::{Duration, Instant},
+    },
+};
+
+pub struct LoadBalancer {
```

Comment:
This should probably be its own PR

Reply:
The StakeUpdater depends on the load balancer to provide a RpcClient to work.

Reply:
I'm not sure what LoadBalancer is supposed to balance exactly. It could use at least some comments explaining its purpose, but also tests to ensure it does what it is supposed to (and also to document in code what it is supposed to do)

Reply:
Update with more comments. I will work on a test setup in the next PR. Thanks.